applications/CompressiblePotentialFlowApplication/custom_operations/potential_to_compressible_navier_stokes_operation.cpp

```diff
@@ -0,0 +1,118 @@
+    const std::string destination_model_part_name = mParameters["destination_model_part"].GetString();
+    const double reference_temperature = mParameters["reference_temperature"].GetDouble();
+    const bool compute_nodal_velocities = mParameters["compute_nodal_velocities"].GetBool();
+    
+    // Saving the modelparts
+    auto& r_origin_model_part = mpModel->GetModelPart(origin_model_part_name);
+    auto& r_destination_model_part = mpModel->GetModelPart(destination_model_part_name);
+
+    const int n_orig_nodes = mOriginModelPart.NumberOfNodes();
+    const int n_dest_nodes = mDestinationModelPart.NumberOfNodes();
+
+    // Check number of nodes
+    KRATOS_ERROR_IF_NOT(n_orig_nodes == n_dest_nodes)
+        << "Origin and destination model parts have different number of nodes."
+        << "\n\t- Number of origin nodes: " << n_orig_nodes
+        << "\n\t- Number of destination nodes: " << n_dest_nodes << std::endl;
+
+    // Getting the required variables
+    const double gamma = mOriginModelPart.GetProcessInfo().GetValue(HEAT_CAPACITY_RATIO);
+    const double sound_velocity = mOriginModelPart.GetProcessInfo().GetValue(SOUND_VELOCITY);
+    const double free_stream_rho = mOriginModelPart.GetProcessInfo().GetValue(FREE_STREAM_DENSITY);
+    const double free_stream_mach = mOriginModelPart.GetProcessInfo().GetValue(FREE_STREAM_MACH);
+    const double specific_heat = (std::pow(sound_velocity,2) / (gamma * reference_temperature)) / (gamma - 1.0);
+    
+    // Calculate de nodal velocities
+    if (compute_nodal_velocities){
+        // Construct the ComputeNodalPotentialFlowVelocityProcess
+        const std::vector<std::string> variable_array = {"VELOCITY"};
+        ComputeNodalValueProcess ComputeNodalValueProcess(mOriginModelPart, variable_array);
+
+        // Execute the ComputeNodalPotentialFlowVelocityProcess
+        ComputeNodalValueProcess.Execute();}
+
+    // Transfer the potential flow values as initial condition for the compressible problem
+    IndexPartition<std::size_t>(n_orig_nodes).for_each([&](std::size_t index)
+                                                       {
+        auto it_dest_node = mDestinationModelPart.NodesBegin() + index;
+        const auto it_orig_node = mOriginModelPart.NodesBegin() + index;
+
+        // Getting potential flow velocities
+        const auto &velocity = it_orig_node->GetValue(VELOCITY);
+        
+        // Calculate the conservative variables
+        const double& velocity_norm_2 = velocity[0] * velocity[0] + velocity[1] * velocity[1];
+        const double& velocity_norm = std::pow(velocity_norm_2,0.5);
+        const double& mach = velocity_norm / sound_velocity;
+        const double& num = 1.0 + 0.5 * (gamma - 1.0) * std::pow(free_stream_mach,2);
+        const double& den = 1.0 + 0.5 * (gamma - 1.0) * std::pow(mach,2);
+        const double& density = free_stream_rho * std::pow((num / den),(1.0 / (gamma - 1.0)));
```

Comment:
Two comments here:

- Try to refer to variables as what they are rather than their greek letter: rho->density, gamma->heat_capacity_ratio, as these letters could have a different meaning in other fields.

- You could use the available utilities to compute the density (they are templated with number of nodes and dimension) and simplify your code. Example of use: https://github.com/KratosMultiphysics/Kratos/blob/0783f3c772c8f19bdaf3f8b1ae52a58ebf62562b/applications/CompressiblePotentialFlowApplication/custom_elements/compressible_potential_flow_element.cpp#L195-L197
Implementation of ComputeDensity:
https://github.com/KratosMultiphysics/Kratos/blob/0783f3c772c8f19bdaf3f8b1ae52a58ebf62562b/applications/CompressiblePotentialFlowApplication/custom_utilities/potential_flow_utilities.cpp#L685-L709